osxcollector/output_filters/opendns/lookup_domains.py

```diff
@@ -30,56 +36,188 @@ def _lookup_iocs(self, all_iocs):
-                0 == len(categorized_info.get('security_categories', []))):
+        elif 0 == status and 0 == len(content_categories) and 0 == len(security_categories):
             return True
         return False
 
-    def _should_store_ioc_info(self, categorized_info, security):
+    def _is_security_info_suspicious(self, security_info):
+        """Analyzes info from OpenDNS and makes a boolean determination of suspicious or not.
+
+        Either looks for low values for a specific set of properties, looks for known participation in
+        a threat campaign, or looks for unknown domains.
+
+        Args:
+            security_info: The result of a call to the security endpoint
+        Returns:
+            boolean
+        """
+        # Categorization of site
+        if any([security_info.get(key, None) for key in self.SECURITY_BAD_KEYS]):
+            return True
+
+        for security_check in self.SECURITY_CHECKS:
+            if security_info.get(security_check.key, security_check.max) <= security_check.threshold:
+                return True
+
+        if not security_info.get('found', False):
+            return True
+
+        return False
+
+    def _should_store_ioc_info(self, category_info, security_info):
         """Figure out whether the data gathered is interesting enough to store in the output.
 
         Args:
-            categorized_info: A dict of info returned by the OpenDNS categorization call
-            security: A dict of info returned by the OpenDNS security call
+            category_info: A dict of info returned by the OpenDNS categorization call
+            security_info: A dict of info returned by the OpenDNS security call
         Returns:
             boolean
         """
-        return categorized_info['is_suspicious'] or security['is_suspicious']
+        return category_info['suspicious'] or security_info['suspicious']
+
+    def _trim_security_result(self, security_info):
+        """Converts the results of a security call into a smaller dict.
+
+        Args:
+            security_info: The result of a call to the security endpoint.
+        Returns:
+            A dict
+        """
+        # dga_score sometimes has the wrong sign, fix that please
+        dga_score = security_info.get('dga_score', 0)
+        if dga_score > 0:
+            security_info['dga_score'] = -1 * dga_score
+
+        # There's a lot of info in the security_info, trim it
+        result = {}
+        for security_check in self.SECURITY_CHECKS:
+            if security_check.key in security_info:
+                result[security_check.key] = security_info[security_check.key]
+        for key in self.SECURITY_BAD_KEYS:
+            if key in security_info:
+                result[key] = security_info[key]
+
+        result['found'] = security_info.get('found', False)
+
+        return result
+
+    # Domain categories to consider suspicious
+    SUSPICIOUS_CATEGORIES = [
```

Comment:
I recommend putting all of these static lists/objects from (SUSPICIOUS_CATEGORIES thru SECURITY_BAD_KEYS) at the beginning of the class (LookupDomainsFilter) rather than at the bottom.